libs/core/langchain_core/tools/base.py

```diff
@@ -208,16 +231,19 @@ def create_schema_from_function(
     Returns:
         A pydantic model with the same arguments as the function.
     """
-    # https://docs.pydantic.dev/latest/usage/validation_decorator/
-    with warnings.catch_warnings():
-        # We are using deprecated functionality here.
-        # This code should be re-written to simply construct a pydantic model
-        # using inspect.signature and create_model.
-        warnings.simplefilter("ignore", category=PydanticDeprecationWarning)
-        validated = validate_arguments(func, config=_SchemaConfig)  # type: ignore
-
     sig = inspect.signature(func)
 
+    if _function_annotations_are_pydantic_v1(sig):
```

Comment:
What happens if there's mixed v1 and v2 namespaces? I'd love to raise an assertion error here if it doesn't work and be explicit given that this code is in core